src/components/molecules/NHSNotifyAuthenticator/NHSNotifyAuthenticator.tsx

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable @typescript-eslint/no-require-imports,unicorn/prefer-module */
+
+import { Authenticator, useAuthenticator } from '@aws-amplify/ui-react';
+import { usePathname, redirect } from 'next/navigation';
+import { ReactNode, FC } from 'react';
+import { Amplify } from 'aws-amplify';
+import { getAmplifyOutputs } from '@utils/get-amplify-outputs';
+
+Amplify.configure(getAmplifyOutputs(), { ssr: true });
+
+type NHSNotifyAutheticatorProps = {
+  children: ReactNode;
+};
+
+const isPathnameExempt = (pathname: string) => {
+  return (
+    pathname.includes('auth') ||
```

Comment:
This seems like it needs some abstraction and encapsulation, so that the page itself defines if it is exempt.

Reply:
i think that nextjs is asking us to make some tradeoffs here - if we want to ensure that by default a page is always protected by the authenticator, we need to put it in the layout code like this, where we do not have access to information about individual pages.

we can move the call to NHSNotifyAuthenticator outside of the layout code into specific page code, but then that comes with the obligation to remember to put it in every new page which i thought was less ideal than specifying exempt pages here. what do you think?

Reply:
I'd go with default of "authenticated" and explicit "public" so that most of the time you don't need to do anything special.  Also happy to iterate on this, it's not a blocker.